### PR TITLE
[lldb][test] Set target OS for API tests in case of remote testing

### PR DESCRIPTION
**Pull request by dzhidzhoev**
Makefile.rules uses HOST_OS and OS variables for determining host and target OSes for API tests compilation.

When lldb's target is set to remote-linux, Makefile.rules script should be executed with the target OS variable set to Linux.

This is useful for the case of Windows-to-Linux cross-testing.

### CODE REVIEW

File: lldb/packages/Python/lldbsuite/test/lldbplatformutil.py [hunk 1]
```diff
@@ -97,6 +101,9 @@ def finalize_build_dictionary(dictionary):
             dictionary = {}
         dictionary["OS"] = "Android"
         dictionary["PIE"] = 1
+    elif target_is_remote_linux():
+        dictionary = dictionary or {}
```
Comment (bulbazord) [inline lldb/packages/Python/lldbsuite/test/lldbplatformutil.py:105]: Suggestion: Pull the dictionary checking-logic out of these blocks and put it at the top of the function:
```
def finalize_build_dictionary(dictionary):
    if dictionary is None:
        dictionary = {}
    if target_is_android():
    # ...
````

File: lldb/packages/Python/lldbsuite/test/lldbplatformutil.py [hunk 2]
```diff
@@ -97,6 +101,9 @@ def finalize_build_dictionary(dictionary):
             dictionary = {}
         dictionary["OS"] = "Android"
         dictionary["PIE"] = 1
+    elif target_is_remote_linux():
```
Comment (labath) [inline lldb/packages/Python/lldbsuite/test/lldbplatformutil.py:104]: For consistency, I think we should do this for all linux systems, not just remote ones. (Ideally, I would do it for all systems overall, but I can understand if you're reluctant to do that.)

You can use `getPlatform` to get the platform/os name for both local and remote runs.

File: lldb/packages/Python/lldbsuite/test/lldbplatformutil.py [hunk 3]
```diff
@@ -56,6 +56,10 @@ def target_is_android():
     return configuration.lldb_platform_name == "remote-android"
 
 
+def target_is_remote_linux():
+    return configuration.lldb_platform_name == "remote-linux"
+
+
```
Comment (labath) [inline lldb/packages/Python/lldbsuite/test/lldbplatformutil.py:62]: I guess this isn't used anymore.

File: lldb/packages/Python/lldbsuite/test/lldbplatformutil.py [hunk 4]
```diff
@@ -92,11 +96,22 @@ def match_android_device(device_arch, valid_archs=None, valid_api_levels=None):
...
+        dictionary["OS"] = "Darwin"
+    else:
+        # Provide uname-like platform name
+        platform_name_to_uname = { "linux": "Linux",
+                "netbsd": "NetBSD",
+                "freebsd": "FreeBSD",
+                "windows": "Windows_NT",
+                }
+        dictionary["OS"] = platform_name_to_uname[getPlatform()]
+        dictionary["HOST_OS"] = platform_name_to_uname[getHostPlatform()]
```
Comment (labath) [inline lldb/packages/Python/lldbsuite/test/lldbplatformutil.py:114]: Why is this under the `else:` branch? I would expect we need to set `HOST_OS` even if we're targetting android or darwin...

**Comment by dzhidzhoev** [comment]
Thank you! Updated it: improved code for Darwin host detection, and removed overriding of HOST_OS in Makefile.

File: lldb/packages/Python/lldbsuite/test/lldbplatformutil.py [hunk 5]
```diff
@@ -113,6 +130,10 @@ def _get_platform_os(p):
             platform = "openbsd"
         return platform
 
+    # Triple is not available if we're not connected yet
+    if p.GetName() == "remote-linux":
+        return "linux"
+
```
Comment (labath) [inline lldb/packages/Python/lldbsuite/test/lldbplatformutil.py:136]: Can you explain why was this necessary? I get the connected part, but who was asking for the platform before connecting?

Comment (dzhidzhoev) [reply lldb/packages/Python/lldbsuite/test/lldbplatformutil.py:136]: That may be relevant if _get_platform_os is called before the connection is established, or, for example, if it has failed to be established.
But yeah, currently I'm not aware of such use cases. Should it be removed?

Comment (labath) [reply lldb/packages/Python/lldbsuite/test/lldbplatformutil.py:136]: Yes, I would very much prefer that, as the code you added only handles linux. To support other oses, we would need to create a whole new mapping from lldb platform names to os strings. That list could then get out of sync, etc., so it would be best to just avoid doing that.

I'm not worried about the debug target going down during testing, as that means the test results will be meaningless anyway. If we find out we regularly need the os name before connecting, then I think we ought to revisit the mechanism for determining the OS.

Comment (dzhidzhoev) [reply lldb/packages/Python/lldbsuite/test/lldbplatformutil.py]: Fixed.

**Review by labath — APPROVED** [review]
Thank you.
